chopper_generator/lib/src/generator.dart

```diff
@@ -298,33 +299,44 @@ final class ChopperGenerator
       bool hasBody = body.isNotEmpty || fields.isNotEmpty;
       if (hasBody) {
         if (body.isNotEmpty) {
+          final DartType bodyType = m.parameters
+              .firstWhere((p) => _typeChecker(chopper.Body).hasAnnotationOf(p))
+              .type;
+          final mapExpression = (formUrlEncoded &&
+                  _isMap(bodyType) &&
+                  !_isMapStringString(bodyType))
+              ? _generateMapToStringExpression(refer(body.keys.first))
+              : refer(body.keys.first);
           blocks.add(
-            declareFinal(Vars.body.toString())
-                .assign(refer(body.keys.first))
-                .statement,
+            declareFinal(Vars.body.toString()).assign(mapExpression).statement,
           );
         } else {
           blocks.add(
             declareFinal(Vars.body.toString())
-                .assign(_generateMap(fields))
+                .assign(_generateMap(fields, isStringValue: formUrlEncoded))
                 .statement,
           );
         }
       }
 
       final bool hasFieldMap = fieldMap.isNotEmpty;
       if (hasFieldMap) {
+        final DartType fieldMapType = m.parameters
+            .firstWhere(
+                (p) => _typeChecker(chopper.FieldMap).hasAnnotationOf(p))
+            .type;
```

Comment:
Doesn't the `fieldMap` variable also exposed the `DartType`? Via `
fieldMap.values.first.typeValue;`?

Reply:
> Doesn't the `fieldMap` variable also exposed the `DartType`? Via ` fieldMap.values.first.typeValue;`?

i tried it but get error: `FormatException: Not an instance of Type.`

```dart
        try{
          _logger.info("fieldMapType = ${fieldMap.values.first.typeValue.toString()}");
        } catch (e) {
          _logger.info("error fieldMapType  \n${e}");
        }
```

run and output :
```dart
[INFO] Chopper Generator:error fieldMapType
FormatException: Not an instance of Type.
```

Reply:
in line 515. ConstantReader receive the  Annotation,  it is not a DartType，so it will crash when call `typeValue`. 
``` dart
    for (final ParameterElement p in method.parameters) {
      final DartObject? a = _typeChecker(type).firstAnnotationOf(p);
      if (annotation != null && a != null) {
        throw Exception(
          'Too many $type annotation for \'${method.displayName}\'',
        );
      } else if (annotation == null && a != null) {
        annotation = a;
        name = p.displayName;
      }
    }

    return annotation == null ? {} : {name: ConstantReader(annotation)};
  }
```

Reply:
Ok was just wondering. Thanks for looking into it!